hs_composite_resource/tests/test_composite_resource.py

```diff
@@ -293,6 +293,72 @@ def test_aggregation_folder_delete(self):
         # there should be 1 GenericLogicalFile objects
         self.assertEqual(GenericLogicalFile.objects.count(), 1)
 
+    def test_delete_folder_1(self):
+        """Here we are testing when a folder is deleted at the root level, no aggregations/files at the
+        same location (root level) get deleted."""
+
+        self.create_composite_resource()
+        # create a folder that has the same name as the file that we will be uploading later for creating a single file
+        # aggregation
+        new_folder = self.generic_file_name.split('.')[0]
+        ResourceFile.create_folder(self.composite_resource, new_folder)
+        # add a generic file type
+        self.add_file_to_resource(file_to_add=self.generic_file)
+
+        # there should be no GenericLogicalFile objects
+        self.assertEqual(GenericLogicalFile.objects.count(), 0)
+
+        txt_res_file = [f for f in self.composite_resource.files.all()
+                        if f.extension == ".txt"][0]
+
+        # set generic logical file
+        GenericLogicalFile.set_file_type(self.composite_resource, self.user, txt_res_file.id)
+        txt_res_file = [f for f in self.composite_resource.files.all()
+                        if f.extension == ".txt"][0]
```

Comment:
Is there a need to get txt_res_file again since it is already defined on line 311?

Reply:
It is needed to get a fresh file from DB after the logical file is created otherwise the check on line 318 will fail. I guess one can instead do `txt_res_file.refresh_from_db()`. Will make this change.